docs/overview.md

```diff
@@ -0,0 +1,797 @@
+| 1         | 1               | 1               | 0        | 1                  | (B)            |
+| 2         | 2               | 2               | 0        | 1                  | (C)            |
+| 3         | 3               | 3               | 0        | 1                  | (D)            |
+| 4         | 0               | 1               | 1        | 2                  | (A,B)          |
+| 5         | 2               | 4               | 4        | 3                  | (C,(A,B))      |
+| 6         | 3               | 5               | 8        | 4                  | (D,(C,(A,B)))  |
+
+- Index `0` represents a cluster containing clusters with indices `0` and `0`. It has a distance of `0` and contains `1` element. It represents the cluster: `(A)`.
+- Index `1` represents a cluster containing clusters with indices `1` and `1`. It has a distance of `0` and contains `1` element. It represents the cluster: `(B)`.
+- Index `4` represents a cluster containing clusters with indices `0` and `1`. It has a distance of `1` and contains `2` elements. It represents the cluster: `(A,B)`.
+- Index `5` represents a cluster containing clusters with indices `2` and `4`. It has a distance of `4` and contains `3` elements. It represents the cluster: `(C,(A,B))`.
+- Index `6` represents a cluster containing clusters with indices `3` and `5`. It has a distance of `8` and contains `4` elements. It represents the cluster: `(D,(C,(A,B)))`.
+
+We can see that the hierarchical clustering generated by this linkage `(D,(C,(A,B)))` matches the clustering we generated manually in the previous section `(((A,B),C),D)`. The ordering is reversed, but the clustering and meaning is the same.
+
+### Generating an Address from the Complete-Linkage Hierarchical Clustering
+
+Once the hierarchical clustering (i.e. linkage) is generated, we can generate flat clusters by choosing a distance threshold, traversing the hierarchical cluster until we find a cluster with a distance that meets our threshold, assigning everything within it to the same flat cluster with the same flat cluster label, and repeating this process until everything in the hierarchical cluster has been flat clustered with a label.
+
+We repeat this process for each threshold, which will use a different distance each time, but use the same hierarchical clustering for each threshold. Each threshold will generate a label and we join the labels together with a delimiter (ex: `.`) to create a flat cluster address.
+
+For example, if our thresholds are `5,3,0`, we start with the first threshold (`5`) and traverse the linkage (the linkage table above) from the root until we find a node where the distance is no more than our threshold. The root is the last element in the linkage (index `6`).
+
+- Index `6` contains `A,B,C,D` and has a distance of `8`, which is greater than our threshold (`5`), so we cannot create one flat cluster with everything. We then look at the two children of this node: index `3` and `5`.
```

Comment:
I think it would help visualize things if you include the snippet of the table you are referring too for each of these step descriptions. For example:

| Index     | Cluster Index 1 | Cluster Index 2 | Distance | Number of Elements | Representation |
| --------- | --------------- | --------------- | -------- | ------------------ | -------------- |
| 6         | 3               | 5               | 8        | 4                  | (D,(C,(A,B)))  |

Reply:
I've tried to improve the clarity here: f873dc2